vaadin-login-flow-parent/vaadin-login-flow/src/main/java/com/vaadin/flow/component/login/LoginOverlay.java

```diff
@@ -122,10 +127,22 @@ private UI getCurrentUI() {
     private void ensureAttached() {
         if (getElement().getNode().getParent() == null) {
             UI ui = getCurrentUI();
-            ui.beforeClientResponse(ui, context -> {
-                ui.addToModalComponent(this);
-                autoAddedToTheUi = true;
```

Comment:
Could this be fixed by wrapping this inside `if (this.isAttached()) { ... }`?

Reply:
Using that check also works for this case but causes another test to fail. I will investigate that.

Reply:
It fails for the self-attach test case, because at the time the this block is executed, the overlay is not attached